source/extensions/filters/http/cache/cache_filter.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/extensions/filters/http/cache/v3alpha/cache.pb.h"
+
+#include "common/common/logger.h"
+
+#include "extensions/filters/http/cache/http_cache.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+/**
+ * A filter that caches responses and attempts to satisfy requests from cache.
+ * It also inherits from std::enable_shared_from_this so it can pass shared_ptrs to async methods,
+ * to ensure that it doesn't get destroyed before they complete.
+ */
+class CacheFilter;
+using CacheFilterSharedPtr = std::shared_ptr<CacheFilter>;
+class CacheFilter : public Http::PassThroughFilter,
+                    public Logger::Loggable<Logger::Id::cache_filter>,
+                    public std::enable_shared_from_this<CacheFilter> {
+public:
+  // Throws EnvoyException if no registered HttpCacheFactory for config.typed_config.
+  static CacheFilterSharedPtr
```

Comment:
This needs to throw during config load, not during filter creation, so you will need to add some type of pre-config check and throw there.

Reply:
Done.